### DISCUSSION TITLE
SSD1306 displays in various sizes

### DISCUSSION THREAD

**Discussion by Banana71** [probonopd/MiniDexed#656] [Show and tell]
Today I tested a 2.42" display "4 pin I2C white" and it works. 
On the photo from top to bottom: 
- SSD1306  0.96’ OLED 128x32
- 1.54 zoll oled modul 1.54 "128x64,  [as plamikcho has already described it here](https://github.com/probonopd/MiniDexed/discussions/279#discussioncomment-8264855)
- 2.42 zoll oled 128x64,  [I bought the display from Aliexpress](https://de.aliexpress.com/item/1005002766729064.html) 
It's important to take the 4 pin I2C Version
![SSD1309](https://github.com/probonopd/MiniDexed/assets/104296205/0bbcffe4-3438-4ece-8225-cec950a3c8ae)

**Comment by diyelectromusic** [comment]
I don't think we've mentioned it anywhere, but it should also now support ST7789 based displays too, although because there are quite a large number of variants, I wouldn't like to suggest it will support them all! :)

Kevin

**Comment by Kirtai** [comment]
That could be very handy for those of us with old eyes :)

**Comment by shawnpinciara** [comment]
@Banana71 can you send the minidexed.ini file with your current configuration? I'm trying with the same 128x64 display (form another vendor i guess: https://www.aliexpress.com/item/4001028369082.html?spm=a2g0o.order_list.order_list_main.26.43b91802OgJweZ) but it doesn't work. plz help!

**Reply by diyelectromusic** [reply]
The biggest issue I find with these displays is some have VCC/GND swapped and some have SCL/SDA swapped, so I'm sure you are, but just in case - do go by the labels on the display and not any diagrams for connecting it up.

The other thing to watch out for isI2C has 7-bit addresses, but the lowest bit is a read/write bit.  So the address is often 0x78>>1 = 0x3C.

Kevin

**Reply by shawnpinciara** [reply]
Thanks Kevin! now it works... also to mention LCDEnabled=1 using Oled too, which i was missing.

**Reply by Banana71** [reply]
Hi @shawnpinciara,
For the sake of completeness, here is an excerpt from my minidexed.ini
```
# HD44780 LCD
LCDEnabled=1
LCDPinEnable=17
LCDPinRegisterSelect=4
LCDPinReadWrite=0
LCDPinData4=22
LCDPinData5=23
LCDPinData6=24
LCDPinData7=25
LCDI2CAddress=0x00

# SSD1306 LCD
# For a 128x32 display, set LCDColumns=20; LCDRows=2
# For a 128x64 display, set LCDColumns=20; LCDRows=4
SSD1306LCDI2CAddress=0x3c
SSD1306LCDWidth=128
SSD1306LCDHeight=64
SSD1306LCDRotate=0
SSD1306LCDMirror=0

# Default is 16x2 display (e.g. HD44780)
LCDColumns=20
LCDRows=2
```

**Comment by GordonGunzelman** [comment]
I have one of these 2.42 inch displays and it works perfectly on MiniDexed (and mini-jv880pi).  Picture is side by side with the 0.96" display.  Just paralled the wiring form the .96" to the 2.42. No change in the software at all.  (I may soon have a couple of the 0.96" displays replaced. I can actually read the 2.42 display while playing.
![1000011544](https://github.com/user-attachments/assets/a2648c56-3341-4111-85b2-9ed7498cb8f5)

**Reply by Kirtai** [reply]
Very cool. I've been thinking of trying one of those displays for the same reason. Glad to see they do actually work :)

**Reply by GordonGunzelman** [reply]
A quick (and dirty) shortcut to a more readable 0.96" dispay is the set SSD1306LCDHeight to 32  in minidexed.ini.  This causes the characters to be twice as tall as usual and makes it easier to read. Look at the left display in the picture.
Gordon

**Reply by probonopd** [reply]
Nice find!

**Reply by Kirtai** [reply]
Just wondering but are those two displays daisy-chained?

**Comment by Kirtai** [comment]
For those people who want to use these OLED displays in a rack mount MiniDexed but find they don't fit, I found [these](https://www.aliexpress.com/item/1005006183194739.html) 2.23" 128x32 screens that look suitable.
They use SSD1305 drivers but those are supposed to work by setting "SSD1306LCDWidth=132".

I will test them shortly, once my current project leaves my workbench :)

Here is a photo of them lying on a 1U half rack/10" rack cardboard template to show the relative sizes.

2.23" 128x32 on the left, 2.24" 128x64 on the right. As you can see, the 2.23" one fits nicely.

![screens](https://github.com/user-attachments/assets/b7aa396f-b2ba-47e2-aadb-d25c9a9fde6c)

**Reply by Kirtai** [reply]
And for comparison, here are a 0.96" OLED on the left and an LCD on the right, on the same template.

![20251011_145102](https://github.com/user-attachments/assets/489e0265-20d1-45a9-8ad3-26281da0f0dc)

**Reply by Kirtai** [reply]
And all together.

![all screens](https://github.com/user-attachments/assets/c22f4491-64a3-48ff-b988-d963246a285a)

I have a 1.52" 128x64 OLED somewhere but it's gone missing.

**Reply by Kirtai** [reply]
I can confirm the 2.23" 128x32 OLEDs I have  work with the following settings
```
SSD1306LCDI2CAddress=0x3C
SSD1306LCDWidth=132
SSD1306LCDHeight=32
```
Screenshot:
![screen 2](https://github.com/user-attachments/assets/ec7ea4f2-b242-464f-a77d-41fc5142a5f4)

Yes, I *am* holding the pins in with my bare hands.

**Reply by Kirtai** [reply]
Found my 1.54" OLED and here's a couple of photos of it next to a 0.96" OLED on the same template.
Both are 128x64.

![screens 1](https://github.com/user-attachments/assets/50635c86-c073-4949-8113-9a615f7b72a5)
![screens 2](https://github.com/user-attachments/assets/9f1aec79-4ac1-4e20-8608-c3bba47ea705)

**Comment by Maximo68** [comment]
My SSD1306 display purchased from Aliexpress isn't working. Unlike others that display GND VCC SCL SDA as the screen print, it reads GND VDD SCK SDA as shown in the photo.
Have you ever encountered a model like this?

![1760191967959](https://github.com/user-attachments/assets/2b7168db-2cff-47f2-9a56-166fbfd3018d)

![1760191967952](https://github.com/user-attachments/assets/733e4128-e724-44b4-b30d-c6f22720de5d)

**Comment by temporal-distortion** [comment]
Hi all,

today i got delivery of an AdaFruit 128x64 oled bonnet (https://learn.adafruit.com/adafruit-128x64-oled-bonnet-for-raspberry-pi?view=all) that i connected to my zero 2 wh and assigned its designated address of 0x3c. The screen worked immediately. Also the buttons are alive, but here i'm unsure how to map them properly (or rather what "pins" this display uses). Anyone here using on of these displays, that i can ask for some advice?

i found this in their article:

# Input pins:
button_A = DigitalInOut(board.D5)
button_A.direction = Direction.INPUT
button_A.pull = Pull.UP

button_B = DigitalInOut(board.D6)
button_B.direction = Direction.INPUT
button_B.pull = Pull.UP

button_L = DigitalInOut(board.D27)
button_L.direction = Direction.INPUT
button_L.pull = Pull.UP

button_R = DigitalInOut(board.D23)
button_R.direction = Direction.INPUT
button_R.pull = Pull.UP

button_U = DigitalInOut(board.D17)
button_U.direction = Direction.INPUT
button_U.pull = Pull.UP

button_D = DigitalInOut(board.D22)
button_D.direction = Direction.INPUT
button_D.pull = Pull.UP

button_C = DigitalInOut(board.D4)
button_C.direction = Direction.INPUT
button_C.pull = Pull.UP


Is it safe to assume that D5, D6, D27, etc corresponds with gpio5, gpio6, gpio27, etc?

**Reply by temporal-distortion** [reply]
Answering this myself for anyone else who has/had the same doubt. 

Yes!

+----------------+-----------+---------------+------------------------+
| Tutorial Name  | BCM GPIO  | Physical Pin  | Function / Direction   |
+----------------+-----------+---------------+------------------------+
| D27            | GPIO 27   | Pin 13        | Joystick Left          |
| D23            | GPIO 23   | Pin 16        | Joystick Right         |
| D17            | GPIO 17   | Pin 11        | Joystick Up            |
| D22            | GPIO 22   | Pin 15        | Joystick Down          |
| D4             | GPIO 4    | Pin 7         | Joystick Center (Push) |
| D5             | GPIO 5    | Pin 29        | Button A               |
| D6             | GPIO 6    | Pin 31        | Button B               |
+----------------+-----------+---------------+------------------------+

**Comment by temporal-distortion** [comment]
Working nicely now with a zero PI 2 wh, adafruid 128x64 and raspiaudio v3 DAC.

Had to struggle a little bit to get all buttons working (joystick has 4 buttons +center click) + 2 separate ones.

<img width="1398" height="1864" alt="image" src="https://github.com/user-attachments/assets/9990857c-d555-4030-9ecb-5e8bad644119" />

Followed banana71s advice here https://github.com/probonopd/MiniDexed/discussions/656#discussioncomment-9608187 about it and it sprung to life perfectly.

Beautiful sound coming out of it as well from the DAC (imho even clearer than the IQAudio DAC+ even).

**Comment by Banana71** [comment]
Could it be that you're the first person to have managed to build a MiniDexed without soldering? What a cool gadget! 🐣

**Reply by temporal-distortion** [reply]
:) I would be honored, humbled and scared if that was the case. No, i can not be the first imho (and not to mention your fantastic sound-sets that comes with this. Purely amazing!)

While i did build a 19-inch rack with physical volume control, etc and soldering cables, etc until my fingers literally bled

<img width="2616" height="1962" alt="image" src="https://github.com/user-attachments/assets/99bf3192-949c-4471-bc9b-a620598f1487" />

<img width="2616" height="1962" alt="image" src="https://github.com/user-attachments/assets/08d30749-7287-42fc-8fa2-5d3f81dbe0f6" />


In this case it was more like making a "sandwich" of sorts... Beautifully sounding sandwich :) And i'm happy how effortless it was.

<img width="2076" height="785" alt="image" src="https://github.com/user-attachments/assets/f8eca3d0-a8be-473a-8a7b-3269668b1ee3" />

**Reply by Banana71** [reply]
The 3-Ply Synth: Extra Soft on your Fingers. No Soldering. No Tears. Just Fun. 🎹

**Comment by FanThomas70** [comment]
Hi, this would be an excellent display for MiniDexed. It has an SSD1322 controller, so it would be necessary to add support for controlling it in the code. I would really like it; they are available at a very favorable price of $20.
![S6f1a203be28a41f2a2e1b3dd5c251f8bz](https://github.com/user-attachments/assets/0167bc75-a055-4864-93f8-34c6b89a925b)
![S016d45dbfa54452fbcd83ed0a992b768g](https://github.com/user-attachments/assets/f7ce4a1b-d2e7-4126-97c0-17517084e601)
![See35fd50ccf34bd39efb8918bc9deed5I](https://github.com/user-attachments/assets/3beca33b-01a7-4177-a0cb-8d2ce434144c)
![S3168c51701024185be85f5809a1fd0e7j](https://github.com/user-attachments/assets/277d806c-5d9e-4663-bb42-f477ee53b680)
![S9d0e16218ea4456885e3e848a827f83a9](https://github.com/user-attachments/assets/f55b2607-19aa-4ed5-91a0-cf886d6ee18d)

**Reply by probonopd** [reply]
It's ~10x as expensive as the ones we are already supporting and it doesn't seem to support 2 lines of text, so I don't see this coming, sorry.

**Reply by craigyjp** [reply]
It is a nice display but it is really expensive, I have a couple at home for testing on another project.  for that money I would buy a genuine 1602 OLED character display. 

A better display is the 2.25" OLED ST7789 colour display, it costs very little, but it's graphic and not character based. 

I just found this on AliExpress: 
SAMIROB New 2.25-inch TFT LCD ST7789 small screen 76*284 module LCD full view SPI color screen
https://a.aliexpress.com/_EIRH248

**Reply by FanThomas70** [reply]
Unfortunately, price is a very important parameter. Right now I’m using a 2.24" 128×64 display because I had it at home from previous projects. Unfortunately, the larger the OLED, the more expensive it is. I wouldn’t mind an IPS or LCD panel either, but at my age I already prefer a physically larger font.

I’m very excited about this project because from the examples it really looks excellent.
